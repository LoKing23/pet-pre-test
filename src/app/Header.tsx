import React from 'react'
import { Flex, Wrap, WrapItem, Avatar, Text } from '@chakra-ui/react';
import CartPopIcon from "./CartPopIcon";

type Props = {
    greeting?: string;
}

const Header = ({
    greeting = 'Good morning!',
}: Props) => {
    return (
        <Flex bgColor="white" boxShadow="xl" p="3" alignItems='center'>
            <Wrap ml="auto">
                <WrapItem>
                    <CartPopIcon />
                </WrapItem>
                <WrapItem>
                    <Text role="name" lineHeight='48px'>
                        { `${greeting} cute cat.` }
                    </Text>
                </WrapItem>
                <WrapItem>
                    <Avatar name='A cute cat' src='https://img.ltn.com.tw/Upload/playing/page/2018/07/08/180708-10152-1-J2OEU.jpg' />
                </WrapItem>
            </Wrap>
        </Flex>
    )
}

export default Header