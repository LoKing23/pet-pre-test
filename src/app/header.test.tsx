import { render, screen } from "@testing-library/react";
import Header from "./header";

describe('Header', () => {

    it('header text correctly', () => {
        render(<Header />);

        expect(screen.getByRole('name').textContent).toBe('Good morning! cute cat.');
    });

    it('header text correctly', () => {
        render(<Header greeting="Good afternoon!" />);

        expect(screen.getByRole('name').textContent).toBe('Good afternoon! cute cat.');
    });

    it('snapshot header', () => {
        const { baseElement } = render(<Header greeting="Good afternoon!" />);

        expect(baseElement).toMatchSnapshot();
    })

    it('snapshot by name role', () => {
        render(<Header greeting="Good afternoon!" />);

        expect(screen.getByRole('name')).toMatchSnapshot();
    })


    it('snapshot text', () => {
        render(<Header greeting="Good afternoon!" />);

        expect(screen.getByRole('name').textContent).toMatchSnapshot();
    })
})