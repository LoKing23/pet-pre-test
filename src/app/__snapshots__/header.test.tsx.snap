// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot by name role 1`] = `
<p
  class="chakra-text css-1orhtrg"
  role="name"
>
  Good afternoon! cute cat.
</p>
`;

exports[`Header snapshot header 1`] = `
<body>
  <div>
    <div
      class="css-14aftgc"
    >
      <div
        class="chakra-wrap css-zdpt2t"
      >
        <ul
          class="chakra-wrap__list css-19lo6pj"
        >
          <li
            class="chakra-wrap__listitem css-1yp4ln"
          >
            <div
              aria-controls="popover-content-:r5:"
              aria-expanded="false"
              aria-haspopup="dialog"
              class="css-12r5c7y"
              id="popover-trigger-:r5:"
            >
              <button
                aria-label="cart button"
                class="chakra-button css-p7mkev"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="22px"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="22px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
                  />
                </svg>
              </button>
              <div
                class="css-1v0wv12"
              >
                0
              </div>
            </div>
            <div
              class="chakra-popover__popper css-0"
              style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
            >
              <section
                aria-describedby="popover-body-:r5:"
                aria-labelledby="popover-header-:r5:"
                class="chakra-popover__content css-yff3ls"
                id="popover-content-:r5:"
                role="tooltip"
                style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
                tabindex="-1"
              >
                <div
                  class="chakra-popover__arrow-positioner css-0"
                  data-popper-arrow=""
                  style="position: absolute;"
                >
                  <div
                    class="chakra-popover__arrow css-0"
                    data-popper-arrow-inner=""
                  />
                </div>
                <button
                  aria-label="Close"
                  class="chakra-popover__close-btn css-6sdaxy"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-icon css-onkibi"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                <header
                  class="chakra-popover__header css-0"
                  id="popover-header-:r5:"
                >
                  購物車
                </header>
                <div
                  class="chakra-popover__body css-0"
                  id="popover-body-:r5:"
                >
                  <p
                    class="chakra-text css-1oxzxkn"
                  >
                    購物車內沒有商品
                  </p>
                </div>
              </section>
            </div>
          </li>
          <li
            class="chakra-wrap__listitem css-1yp4ln"
          >
            <p
              class="chakra-text css-1orhtrg"
              role="name"
            >
              Good afternoon! cute cat.
            </p>
          </li>
          <li
            class="chakra-wrap__listitem css-1yp4ln"
          >
            <span
              class="chakra-avatar css-1ruhx4w"
            >
              <div
                aria-label="A cute cat"
                class="chakra-avatar__initials css-0"
                role="img"
              >
                Ac
              </div>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;

exports[`Header snapshot text 1`] = `"Good afternoon! cute cat."`;
